detectsex.rb

```diff
@@ -0,0 +1,80 @@
+require 'telegram/bot'
+require 'net/http'
+require 'json'
+require 'dotenv'
+
+Dotenv.load('.env')
+TGTOKEN = ENV['TGTOKEN']
+MSTOKEN = ENV['MSTOKEN']
+
+FACE_API_URI = 'https://westcentralus.api.cognitive.microsoft.com/face/v1.0/detect'.freeze
+API_TELEGRAM = 'https://api.telegram.org/'.freeze
+
+def prepare_url(params)
+  API_TELEGRAM + params
+end
+
+def prepare_answer(answer_from_faceapi)
+  if answer_from_faceapi.count > 1
+    'Дружище, на фото больше одного человека, если хочешь, чтобы я рассказал тебе о каждом из них - тебе придётся ждать пока @holywalley меня заапдейтит :)'
+  elsif answer_from_faceapi.count == 1
+    if answer_from_faceapi[0]['faceAttributes']['gender'] == 'male'
+      "Кхм, твои вкусы очень специфичны, ведь на фото мужчина. Кстати, ему #{answer_from_faceapi[0]['faceAttributes']['age']} лет"
+    else
+      "Ей #{answer_from_faceapi[0]['faceAttributes']['age']} лет, не благодари :) \nА хотя, ладно, благодари @holywalley"
+    end
+  else
+    'Ну тут два варианта: либо я не вижу её лицо, либо ты меня обманываешь :('
+  end
+end
+
+def get_image_true_path(message)
+  file_id = message.photo.last.file_id
+
+  uri = URI(prepare_url('bot' + TGTOKEN + "/getFile?file_id=#{file_id}"))
+  res = Net::HTTP.get(uri)
+  my_hash = JSON.parse(res)
+
+  image_true_path = prepare_url('file/bot' + TGTOKEN + "/#{my_hash['result']['file_path']}")
+end
+
+def get_response_from_faceapi(image_true_path)
+  uri = URI(FACE_API_URI)
+  uri.query = URI.encode_www_form(
+    'returnFaceId' => 'true',
+    'returnFaceLandmarks' => 'false',
+    'returnFaceAttributes' => 'age,gender'
+    )
+  request = Net::HTTP::Post.new(uri.request_uri)
+  request['Content-Type'] = 'application/json'
+  request['Ocp-Apim-Subscription-Key'] = MSTOKEN
+  request.body = {:url => image_true_path}
+  request.body = JSON.generate(request.body)
+  response = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') do |http|
+    http.request(request)
+  end
+  response
+end
+
+def send_message(bot, message, s_message)
+  bot.api.send_message(chat_id: message.chat.id, text: s_message)
+end
+
+Telegram::Bot::Client.run(TGTOKEN) do |bot|
+  bot.listen do |message|
+    case message.text
+    when '/start'
+      send_message(bot, message, 'Привет, друг! Хочешь узнать, сколько лет девушке, но напрямую спрашивать невежливо? ОК - скинь её фото и я тебе помогу :)')
+      send_message(bot, message, '/start - моё описание и список команд (будет пополняться)')
+      send_message(bot, message, 'Просто загружай фото и я отвечу на твой вопрос!')
+    end
+    if message.photo.last
+      image_true_path = get_image_true_path(message)
+      puts 'New Photo Uploaded'
+      send_message(bot, message, "Упс, что-то пошло не так :) Попробуйте позже") unless response = get_response_from_faceapi(image_true_path)
```

Comment:
the check is ok, but code will be executed after this check and next line will blow with undefined method on nil :)